llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -7601,16 +7601,20 @@ SDValue SelectionDAG::getNode(unsigned Opcode, const SDLoc &DL, EVT VT,
     case ISD::SDIV:
     case ISD::UREM:
     case ISD::SREM:
-      return getUNDEF(VT);       // fold op(arg1, undef) -> undef
+      // fold op(arg1, undef) -> undef, // fold op(arg1, poison) -> poison.
```

Comment:
Please also update the switch in the else block of the `N1.isUndef()` check on line 7573. We should be consistent about poison on both the left and right hand side for the same opcodes.